corehq/apps/data_cleaning/views/filters.py

```diff
@@ -32,36 +31,36 @@ class PinnedFilterFormView(BulkEditSessionViewMixin, BaseFilterFormView):
     session_not_found_message = gettext_lazy("Cannot retrieve pinned filter, session was not found.")
 
     @property
-    @memoized
     def timezone(self):
         return get_timezone(self.request, self.domain)
 
-    @property
-    def filter_type(self):
-        return self.kwargs['filter_type']
-
-    @property
-    def form_filter_class(self):
-        filter_type_to_class = {
+    @staticmethod
+    def get_form_filter_class(filter_type):
+        return {
             PinnedFilterType.CASE_OWNERS: CaseOwnersPinnedFilter,
             PinnedFilterType.CASE_STATUS: CaseStatusPinnedFilter,
-        }
-        try:
-            return filter_type_to_class[self.filter_type]
-        except KeyError:
-            raise Http404(_("unsupported filter type: {}").format(self.filter_type))
+        }[filter_type]
 
     @property
     @memoized
-    def form_filter(self):
-        return self.form_filter_class(
-            self.request, self.domain, self.timezone, use_bootstrap5=True
-        )
+    def form_filters(self):
+        return [
+            (f.filter_type, self.get_form_filter_class(f.filter_type)(
+                self.request, self.domain, self.timezone, use_bootstrap5=True
+            )) for f in self.session.pinned_filters.all()
+        ]
+
+    @hq_hx_action('post')
+    def update_filters(self, request, *args, **kwargs):
+        # todo
+        return self.get(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context.update({
-            "filter_type": self.filter_type,
-            "rendered_filter": self.form_filter.render(),
+            'container_id': 'pinned-filters',
+            'form_filters': [
+                f[1].render() for f in self.form_filters
```

Comment:
Could `form_filters` be a dict instead of a tuple? It's a little more readable than accessing tuple parts by index.

Reply:
nevermind - this code changed in a later commit

Reply:
i think this comment is not relevant after [this commit](https://github.com/dimagi/commcare-hq/pull/35880/commits/68ecb225332544900098d3f32099aaed7503f07b)... yes?

Reply:
yes